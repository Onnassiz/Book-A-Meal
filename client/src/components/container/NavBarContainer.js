import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import NavBar from '../presentation/layout/NavBar';

const mapStateToProps = (state) => {
	return {
		formState: state.formState,
		user: state.user,
	};
};

function mapDispatchToProps(dispatch) {
	return bindActionCreators({
	}, dispatch);
}

export default connect(
	mapStateToProps,
	mapDispatchToProps,
)(NavBar);
