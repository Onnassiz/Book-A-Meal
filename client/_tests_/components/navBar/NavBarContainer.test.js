import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import configureStore from 'redux-mock-store';
import NavBarContainer from '../../../src/components/container/NavBarContainer';
import userInfo from '../../objectProps/user.props';

const mockStore = configureStore();
const store = mockStore({ user: userInfo });

const props = {
  users: {},
  page: 'home',
};

const wrapper = shallow(<NavBarContainer {...props} store={store} />);

describe('<NavBarContainer />', () => {
  it('Should render container component', () => {
    expect(wrapper.length).toBe(1);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
