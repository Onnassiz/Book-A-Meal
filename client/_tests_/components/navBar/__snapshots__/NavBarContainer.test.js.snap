// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavBarContainer /> Should render container component 1`] = `
<NavBar
  page="home"
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": [Function],
    }
  }
  user={
    Object {
      "email": "onnassiz@gmail",
      "id": "3a3c499f-803a-4e06-a328-307daff22243",
      "name": "Ben Onah",
      "role": "caterer",
    }
  }
  users={Object {}}
/>
`;
